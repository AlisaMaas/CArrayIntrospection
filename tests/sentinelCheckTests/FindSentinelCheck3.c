void print(char* string){}
int find(char string[], int distance)
{
  for(int i = 0; i < distance; i++){
		if(string[i] == '\0'){
			break;
			}
	}
  return 1;
}
