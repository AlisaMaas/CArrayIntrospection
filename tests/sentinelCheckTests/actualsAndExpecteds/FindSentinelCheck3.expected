Printing analysis 'Promote Memory to Register' for function 'print':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'bar':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'foo':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'baz':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'find':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Find each branch used to exit a loop when a sentinel value is found in an array':
Analyzing function: print
	We found: 0 loops
Analyzing function: bar
	We found: 1 loops
	Examining string in loop for.cond
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body
Analyzing function: foo
	We found: 1 loops
	Examining string in loop for.cond
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body
Analyzing function: baz
	We found: 1 loops
	Examining string in loop for.cond
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body
Analyzing function: find
	We found: 14 loops
	Examining string in loop for.cond
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body
	Examining string in loop for.cond109
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body112
	Examining string in loop for.cond124
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body127
	Examining string in loop for.cond139
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body142
	Examining string in loop for.cond154
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body157
	Examining string in loop for.cond169
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body172
	Examining string in loop for.cond184
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body187
	Examining string in loop for.cond19
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body22
	Examining string in loop for.cond34
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body37
	Examining string in loop for.cond4
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body7
	Examining string in loop for.cond49
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body52
	Examining string in loop for.cond64
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body67
	Examining string in loop for.cond79
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body82
	Examining string in loop for.cond94
		There are 1 sentinel checks of this argument in this loop
			We can not bypass all sentinel checks for this argument in this loop.
		Sentinel checks: 
			for.body97
